lib/Doctrine/ORM/Mapping/JoinTableMapping.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ArrayAccess;
+
+use function array_map;
+use function in_array;
+
+/** @template-implements ArrayAccess<string, mixed> */
+final class JoinTableMapping implements ArrayAccess
+{
+    public bool|null $quoted = null;
+
+    /** @var list<JoinColumnMapping> */
+    public array $joinColumns = [];
+
+    /** @var list<JoinColumnMapping> */
+    public array $inverseJoinColumns = [];
+
+    public string|null $schema = null;
+
+    public string|null $name = null;
+
+    /** @param array{name?: string, quoted?: bool, joinColumns?: mixed[], inverseJoinColumns?: mixed[], schema?: string} $mappingArray */
+    public static function fromMappingArray(array $mappingArray): self
+    {
+        $mapping = new self();
+
+        foreach (['name', 'quoted', 'schema'] as $key) {
+            if (isset($mappingArray[$key])) {
+                $mapping[$key] = $mappingArray[$key];
+            }
+        }
+
+        if (isset($mappingArray['joinColumns'])) {
+            foreach ($mappingArray['joinColumns'] as $column) {
+                $mapping->joinColumns[] = JoinColumnMapping::fromMappingArray($column);
+            }
+        }
+
+        if (isset($mappingArray['inverseJoinColumns'])) {
+            foreach ($mappingArray['inverseJoinColumns'] as $column) {
+                $mapping->inverseJoinColumns[] = JoinColumnMapping::fromMappingArray($column);
+            }
+        }
+
+        return $mapping;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function offsetExists($offset): bool
+    {
+        return isset($this->$offset);
+    }
+
+    public function offsetGet($offset): mixed
```

Comment:
Except for `offsetSet` you could use the methods of `ArrayAccessImplementation`. It's possible to override a method of a trait, but I don't know if this is a commendable practice.